### PR TITLE
[#162171] Cross-Core - (2) Add form inputs for adding to an existing order

### PR DESCRIPTION
**Pull request by LeticiaErrandonea**
# Release Notes
* Add field to select facility and update dropdown when adding products to existing order.

# Screenshot
## Facility admin from selected facility
image

## Facility admin from another facility
image

## Facility staff
image

# Additional Context

Optional. Feel free to add/modify additional headers as appropriate, e.g. "Refactorings", "Concerns".

# Accessibility
- [ ] Did you scan for accessibility issues?
- [ ] Did you check our accessibility goal checklist?

### CODE REVIEW

File: app/views/facility_orders/_merge_order_form.html.haml [hunk 1]
```diff
@@ -3,6 +3,12 @@
     = simple_form_for [current_facility, @add_to_order_form], url: facility_order_path(current_facility, @order), method: :put, html: { class: "js--edit-order" } do |f|
       = f.input_field :quantity, class: "js--edit-order__quantity", style: "width: 3em;"
 
+      - unless Rails.env.production? || !can?(:available_for_cross_core_ordering, Product)
```
Comment (giladshanan) [inline app/views/facility_orders/_merge_order_form.html.haml:6]: How about putting this on a separate line, above the quanatity/product inputs?  And let's give this and the product dropdown a label as well

File: spec/system/admin/adding_to_an_order_spec.rb [hunk 2]
```diff
@@ -185,4 +185,34 @@
...
 
+  describe "adding a product from another facility", :js do
+    let(:facility2) { create(:setup_facility) }
+    let(:product) { create(:setup_item, :with_facility_account) }
+    let!(:product2) { create(:setup_item, :with_facility_account, facility: facility2) }
+
+    before do
+      visit facility_order_path(facility, order)
+    end
+
```
Comment (giladshanan) [inline spec/system/admin/adding_to_an_order_spec.rb:196]: How about adding an accessibility scan here:
```
    it "is accessible" do
      expect(page).to be_axe_clean
    end
```
If it passes or there are simple fixes to make it pass, great.  If not, you can limit the scope checked by the spec (see [docs](https://docs.deque.com/devtools-html/4.0.0/en/rb-rspec)) and then we can improve it later on

**Review by giladshanan — CHANGES_REQUESTED** [review]
Nice work!  Some small nitpicks and rearranging of the inputs and this will be ready to go

File: app/views/facility_orders/_merge_order_form.html.haml [hunk 3]
```diff
@@ -3,6 +3,12 @@
...
+        = f.input_field :facility_id,
+          collection: @add_to_order_form.facilities.map { |f| [f.name, f.id, {"data-products-path" => available_for_cross_core_ordering_facility_products_path(f, format: :js)}] },
+          class: "js--edit-order__facility js--chosen",
+          include_blank: false
```
[the remark this answers is not in the document]
Comment (LeticiaErrandonea) [reply app/views/facility_orders/_merge_order_form.html.haml:10]: Yes

[the remark this answers is not in the document]
Comment (LeticiaErrandonea) [reply app/views/facility_orders/_merge_order_form.html.haml:10]: image

**Pull request merged**